core/engine/src/value/inner/nan_boxed.rs

```diff
@@ -296,8 +321,8 @@ mod bits {
     /// by calling `[Self::drop_pointer]`.
     #[inline(always)]
     #[allow(clippy::identity_op)]
-    pub(super) unsafe fn tag_bigint(value: Box<JsBigInt>) -> u64 {
-        let value = Box::into_raw(value) as u64;
+    pub(super) unsafe fn tag_bigint(value: Rc<JsBigInt>) -> u64 {
```

Comment:
Suggestion: Maybe we should pass `JsBigInt` as value directly since it's wrapped in an `Rc<RawBigInt>` already internally, and add methods for `into_raw()`/`from_raw()`.

Reply:
This cannot work unless I change the interface of the JsValue to return `Rc<...>` (or similar) instead of references. To return a `&'a JsBigInt` with your suggested change I'd need to reconstruct it on the stack, and I cannot return a reference to _that_. Unless I'm missing something?

Reply:
I see, sorry I missed that :sweat_smile: , we could get around this by using a bit of safe "unsafe" code by adding `#[repr(transparent)]` on `JsBigInt` and `transmute`-ing `&'a Rc<RawBigInt>` into `&'a JsBigInt` JsBigInt.

For more info see: https://doc.rust-lang.org/nomicon/other-reprs.html#reprtransparent

Reply:
I kind of tried that. I need to spend more time on it, but I was still getting segfaults accessing freed or invalid memory.